querydsl-jpa/src/main/java/com/querydsl/jpa/HibernateHandler.java

```diff
@@ -37,16 +37,18 @@
     @Override
     public void addEntity(Query query, String alias, Class<?> type) {
         if (query instanceof NativeQuery) {
-            NativeQuery hibernateQuery = (NativeQuery) query;
-            hibernateQuery.addEntity(alias, type);
+            query.unwrap(NativeQuery.class).addEntity(alias, type);
```

Comment:
When `unwrap` is used, it doesnt have to be cast to a `NativeQuery` in the first place. In fact, unwrap was introduced to prevent the need for casts, which is important in i.e. J2EE environments where the `Query` may be wrapped in a proxy that does not implement `NativeQuery`.

`unwrap` will throw a `PersistenceException` if the implementation is not of the requested type. This would be the exception to catch and rethrow in the `IllegalArgumentException`. Or just let it be a `PersistenceException`.

Reply:
I think it sounds better to just keep it `PersistenceException` rather than rethrow.

Reply:
Thanks for the review :) 👍 

Of the two suggested, I think it is better to throw PersistenceException.
So I just removed the conditional statement, and pushed new commit. :)